client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -20,6 +20,32 @@ bool normalTexturePresent = false;
 const float e = 2.718281828459;
 const float BS = 10.0;
 
+#ifdef ENABLE_TONE_MAPPING
+const float A = 0.22;
+const float B = 0.30;
+const float C = 0.10;
+const float D = 0.20;
+const float E = 0.01;
+const float F = 0.30;
+const float W = 11.2;
+
+vec3 uncharted2Tonemap(vec3 x)
+{
+	return ((x * (A * x + C * B) + D * E) / (x * (A * x + B) + D * F)) - E / F;
```

Comment:
..I don't suppose there's any way for you to describe the black magic going on here, is there?